src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -349,8 +351,13 @@ export class StudyViewPageStore {
     }
 
     @action
-    updateGeneFilters(entrezGeneIds: number[]) {
-        this._mutatedGeneFilter = _.xor(this._mutatedGeneFilter, entrezGeneIds);
+    addGeneFilters(entrezGeneIds: number[]) {
+        this._mutatedGeneFilter = _.reduce(this._mutatedGeneFilter, function (acc, next) {
```

Comment:
no need to iterate over the list here, may be `this._mutatedGeneFilter = [...this._mutatedGeneFilter, {entrezGeneIds: entrezGeneIds}]`